### DISCUSSION TITLE
Query devices filtering by tag

### DISCUSSION THREAD

**Discussion by alexgumo7** [CrowdStrike/falconpy#566] [Q&A] [answered]
Hello,

I'm testing the different available options for searching hosts by using [QueryDevicesFilter](https://github.com/CrowdStrike/falconpy/wiki/Hosts#querydevicesbyfilter), with the [filters](https://github.com/CrowdStrike/falconpy/wiki/Hosts#available-filters) available.

The most useful parameter for me at the moment are the tags. Using the console I can add/remove `Grouping Tags` and search the hosts associated to the tags by using the `Grouping Tags` filter. However, in the API I've found the option of adding/removing tags with [UpdateDeviceTags](https://github.com/CrowdStrike/falconpy/wiki/Hosts#updatedevicetags) but when searching hosts with QueryDevicesFilter using the API I'm not able to find a tag filter.

Could you point me how to filter by tag using the API or this option is not included?

Thanks!

**Comment by jshcodes** [comment, marked as answer]
Hi @alexgumo7!

Give this a shot and let us know if you have any questions or problems.

```python
from falconpy import Hosts

hosts = Hosts(client_id="CLIENT_ID_HERE", client_secret="CLIENT_SECRET_HERE")

results = hosts.query_devices_by_filter(filter="tags:['FalconGroupingTags/testtag']")
```

**Comment by alexgumo7** [comment]
Thanks @jshcodes! It works perfectly.
